HOWTO-make-a-release.md

```diff
@@ -0,0 +1,374 @@
+    -   [Software](#software)
+    -   [Repositories](#repositories)
+    -   [PGP / GnuPG key](#pgp-gnupg-key)
+    -   [SSH access](#check-your-access)
+    -   [A method for reviewing](#a-way-to-reviewing)
+-   [Pre-publishing tasks](#pre-publishing-tasks)
+    -   [Freeze the source repository](#freeze-the-source-repository) [the day before release]
+    -   [Prepare your repository checkouts](#prepare-your-repository-checkouts)
+    -   [Make sure that the openssl source is up to date](#make-sure-that-the-openssl-source-is-up-to-date)
+    -   [Generate the tarball and announcement text](#generating-the-tarball-and-announcement-text) [do not push]
+        -   [OpenSSL 3.0 and on](#openssl-3.0-and-on)
+        -   [OpenSSL before 3.0](#openssl-before-3.0)
+    -   [Update the website locally](#update-the-website-locally) [do not push]
+-   [Publish the release](#publish-the-release)
+-   [Post-publishing tasks](#post-publishing-tasks)
+    -   [Check the website](#check-the-website)
+    -   [Send the announcement mail](#send-the-announcement-mail)
+    -   [Unfreeze the source repository](#unfreeze-the-source-repository)
+    -   [Security fixes](#security-fixes)
+    -   [Keep in touch](#keep-in-touch)
+
+
+# Prerequisites
+
+## Software
+
+Apart from the basic operating system utilities, you must have the following
+programs in you `$PATH`:
+
+- openssl
+- ssh
+- gpg
+- git
+
+(note: this may not be a complete list)
+
+## Repositories
+
+You must have access to the following repositories:
+
+-   `openssl-git@git.openssl.org:openssl.git`
+
+    This is the usual main source repository
+
+-   `openssl-git@git.openssl.org:openssl-web.git`
+
+    This is the website repository
+
+-   `openssl-git@git.openssl.org:tools.git`
+
+    This contains certain common tools
+
+## PGP / GnuPG key
+
+You must have a PGP / GnuPG key, and its fingerprint should be present
+in the file `doc/fingerprints.txt` in the source of the immediately prior
+OpenSSL release.
+
+## SSH access
+
+To perform a release, you must have appropriate access to OpenSSL's
+development host, dev.openssl.org.  To test this, try to log in with
+ssh:
+
+    ssh dev.openssl.org
+
+You must also check that you can perform tasks as the user 'openssl'
+on dev.openssl.org.  When you have successfully logged in, test your
+access to that user with sudo:
+
+    sudo -u openssl id
+
+## A method for reviewing
+
+For reviewing to take place, the release person and the reviewer need a
+way to share changes that are being applied.  Most commonly, that's done
+as PRs (for normal releases) or security advisories (for undisclosed
+security fixes) through Github.
+
+Security advisories are created using the Github Security tab, and
+will generate a private branch, ready to be pushed to.  For more
+information on creating new security advisories, please read
+<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/creating-a-security-advisory>
```

Comment:
It would be nice if you could also adopt the [new markdown link style] which I've been advertising recently to aid my goal of providiing high quality [minimalistic markdown] documentation which looks great both in ASCII and when rendered in HTML. 

[new markdown link style]: <https://github.com/openssl/openssl/commit/6c8149df1fb6fce50a914a70040955d3512b0bd6>
[minimalistic markdown]:   <https://github.com/openssl/openssl/commit/5f8e6c50bd862d6bb0bbd16b4c0b3be841d9cf32>

_Note: this comment itself was created using minimalistic markdown:_

    It would be nice if you could also adopt the [new markdown link style] which I've been advertising recently to aid the goal of providing high quality [minimalistic markdown] documentation which looks great both in ASCII and when rendered in HTML. 

    [new markdown link style]: 
    <https://github.com/openssl/openssl/commit/6c8149df1fb6fce50a914a70040955d3512b0bd6>
    [minimalistic markdown]:   
    <https://github.com/openssl/openssl/commit/5f8e6c50bd862d6bb0bbd16b4c0b3be841d9cf32>

Reply:
Here's a suggestion, addressing both comments above:

```suggestion
Security advisories are created using the Github Security tab, and
will generate a private repository, which can be used to work on
fixing the issue via pull requests. It is also possible to add
external collaborators to the private repository, for example the
reporters of the issue or third party companies affected by it.
For more information on [creating a security advisory] resp.
[adding collaborators to a security advisory], please consult the
Github help.

[creating a security advisory]:
<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/creating-a-security-advisory>

[adding collaborators to a security advisory]:
<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/adding-a-collaborator-to-a-security-advisory>
```

Which will be rendered as

Security advisories are created using the Github Security tab, and will generate a private repository, which can be used to work on fixing the issue via pull requests. It is also possible to add external collaborators to the private repository, for example the reporters of the issue or third party companies affected by it. For more information on [creating a security advisory] resp. [adding collaborators to a security advisory], please consult the Github help.

[creating a security advisory]:
<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/creating-a-security-advisory>

[adding collaborators to a security advisory]:
<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/adding-a-collaborator-to-a-security-advisory>


While raw links are important for the ASCII version, in the rendered version they should be embedded naturally in the text and the reader should not be distracted by raw links IMHO.

Reply:
I blame @mattcaswell for "private branch" (https://github.com/openssl/tools/pull/75#issuecomment-728954256) :wink:

Sure, I can use the minimalistic links.

Reply:
Fixed it.  Not exactly like you suggested, but I hope I incorporated your thoughts well enough.